src/reducers/scenes/StakingReducer.ts

```diff
@@ -0,0 +1,148 @@
+import { combineReducers, Reducer } from 'redux'
+
+import { StakePlugin, StakePolicy, StakePosition } from '../../plugins/stake-plugins/types'
+import { Action } from '../../types/reduxTypes'
+
+interface AddStakePolicyAction {
+  type: 'STAKING/ADD_POLICY'
+  walletId: string
+  stakePolicy: StakePolicy
+}
+
+interface FinishAction {
+  type: 'STAKING/FINISH_LOADING'
+  walletId: string
+}
+
+interface UpdateLockedAmountAction {
+  type: 'STAKING/UPDATE_LOCKED_AMOUNT'
+  walletId: string
+  lockedNativeAmount: string
+}
+
+interface UpdatePluginsAction {
+  type: 'STAKING/UPDATE_PLUGINS'
+  walletId: string
+  stakePlugins: StakePlugin[]
+}
+
+interface UpdatePoliciesAction {
+  type: 'STAKING/UPDATE_POLICIES'
+  walletId: string
+  stakePolicies: StakePolicyMap
+}
+
+interface UpdatePositionAction {
+  type: 'STAKING/UPDATE_POSITION'
+  walletId: string
+  stakePolicyId: string
+  stakePosition: StakePositionMap[string]
+}
+
+interface UpdatePositionsAction {
+  type: 'STAKING/UPDATE_POSITIONS'
+  walletId: string
+  stakePositionMap: StakePositionMap
+}
+
+export type StakingAction =
+  | AddStakePolicyAction
+  | FinishAction
+  | UpdateLockedAmountAction
+  | UpdatePluginsAction
+  | UpdatePoliciesAction
+  | UpdatePositionAction
+  | UpdatePositionsAction
+
+export interface StakingState {
+  walletStakingMap: WalletStakingStateMap
+}
+
+export interface WalletStakingStateMap {
+  [walletId: string]: WalletStakingState
+}
+
+export interface WalletStakingState {
+  isLoaded: boolean
+  lockedNativeAmount: string
+  stakePlugins: StakePlugin[]
+  stakePolicies: StakePolicyMap
+  stakePositionMap: StakePositionMap
+}
+
+export interface StakePolicyMap {
+  [policyId: string]: StakePolicy
+}
+
+export interface StakePositionMap {
+  [policyId: string]: StakePosition
+}
+
+export const staking: Reducer<StakingState, Action> = combineReducers({
+  walletStakingMap: (state = {}, action: Action) => {
+    switch (action.type) {
+      case 'STAKING/ADD_POLICY':
+      case 'STAKING/FINISH_LOADING':
+      case 'STAKING/UPDATE_LOCKED_AMOUNT':
+      case 'STAKING/UPDATE_PLUGINS':
+      case 'STAKING/UPDATE_POLICIES':
+      case 'STAKING/UPDATE_POSITION':
+      case 'STAKING/UPDATE_POSITIONS':
```

Comment:
Optional: This could be `if (action.type.startsWith('STAKING/'))`, which might be more sustainable. This is the approach the core uses with its wallet actions, for instance.